core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -367,18 +367,16 @@ class CivilizationInfo {
     fun getCivResources(): ResourceSupplyList = summarizedCivResources
 
     // Preserves some origins for resources so we can separate them for trades
-    fun getCivResourcesWithOriginsForTrade(): ResourceSupplyList {
-        val newResourceSupplyList = ResourceSupplyList()
+    fun getCivResourcesWithOriginsForTrade() = ResourceSupplyList(keepZeroAmounts = true).apply {
         for (resourceSupply in detailedCivResources) {
             // If we got it from another trade or from a CS, preserve the origin
-            if ((resourceSupply.origin == "City-States" || resourceSupply.origin == "Trade") && resourceSupply.amount > 0) {
-                newResourceSupplyList.add(resourceSupply.resource, resourceSupply.amount, resourceSupply.origin)
-                newResourceSupplyList.add(resourceSupply.resource, 0, "Tradable") // Still add an empty "tradable" entry so it shows up in the list
+            if (resourceSupply.isCityStateOrTrade()) {
```

Comment:
The comment I just preserved, not my words. May be another thing to learn - I put a tad too much emphasis on preserving comments when shifting other coder's stuff around.
The name - well since the ResourceSupply only has two keys and one of them clearly fits the name parts... Will reconsider.
Have another '5min quick & nice' ("why does that key not what I expect?") that spilled over under my fingers so it'll be a while to switch branches.

Reply:
I _did_ consider removing the comment and decided to leave both in. Not inappropriate, clear enough without but a tad clearer with, hurts nobody but SSD cells